src/main/java/org/myrobotlab/service/Tracking.java

```diff
@@ -1,592 +1,395 @@
+        // invokeOn(pan, "moveTo", pid.value);
+      } else if (pid.key.equals(tilt)) {
+        send(tilt, "moveIncr", pid.value);
+        // send(tilt, "moveIncr", pid.value);
+        // log.warn("y {}", pid.value);
+        // invokeOn(tilt, "moveTo", pid.value);
+      } else {
+        warn("unknown onPid key %s", pid.key);
+      }
     }
-    opencv.invokeFilterMethod(FILTER_LK_OPTICAL_TRACK, "samplePoint", x, y);
-  }
-
-  public void findFace() {
-    scan = true;
-  }
-
-  public OpenCVData foundFace(OpenCVData data) {
-    return data;
   }
 
-  // TODO - enhance with location - not just heading
-  // TODO - array of attributes expanded Object[] ... ???
-  // TODO - use GEOTAG - LAT LONG ALT DIRECTION LOCATION CITY GPS TIME OFFSET
-  /*
-   * public OpenCVData setLocation(OpenCVData data) {
-   * data.setX(x.getPosition()); data.setY(y.getPosition()); return data; }
+  /**
+   * verify all required service and configuration start tracking
    */
+  public void enable() {
+    if (pid == null) {
+      error("pid not attached");
+      return;
+    }
+    if (pan == null) {
+      error("pan not attached");
+      return;
+    }
+    if (tilt == null) {
+      error("tilt not attached");
+      return;
+    }
+    if (cv == null) {
+      error("cv not attached");
+      return;
+    }
+    // configuring pids
+    // xPid.key = pan;
+    // yPid.key = tilt;
 
-  // ------------------- tracking & detecting methods end
-  // ---------------------
+    // send(pid, "addPid", xPid);
+    // send(pid, "addPid", yPid);
 
-  public OpenCV getOpenCV() {
-    return opencv;
-  }
+    // preserve pre-existing filters ?
+    // put back when disabled ?
 
-  public String getState() {
-    return state;
+    subscribe(pid, "publishPid");
+    subscribe(cv, "publishClassification");
+    for (String filter : trackingFilters) {
+      send(cv, "addFilter", filter);
+    }
+    send(cv, "capture");
+    state = TrackingState.SEARCHING;
+    invoke("publishTrackingState");
+    broadcastState();
   }
 
-  public ServoControl getX() {
-    return servoControls.get("x").servoControl;
+  @Deprecated /* DONT USE THIS ONLY FOR INTERNAL USE ! */
+  public String setState(String state) {
+    try {
+      this.state = TrackingState.valueOf(state.toUpperCase());
+      return state;
+    } catch (Exception e) {
+      error("%s not valid state", state);
+    }
+    return null;
   }
 
-  public Pid getPID() {
-    return pid;
-  }
+  /**
+   * FIXME - should be simple (non opencv) data here as a parameter FIXME -
+   * rename onClassification to onTracked - not classification specific
+   * publishDetected boundingbox + label
+   * 
+   * publishClassifications
+   * 
+   * @param data
+   */
+  // public void onOpenCVData(OpenCVData data) {
+  //
+  // }
 
-  public ServoControl getY() {
-    return servoControls.get("y").servoControl;
-  }
+  public void onClassification(Map<String, List<Classification>> data) {
+    stats.inputCnt++;

```

Comment:
not all opencv filters publish this method signature.. if you start the dl4j filter, it causes this method to blow up.